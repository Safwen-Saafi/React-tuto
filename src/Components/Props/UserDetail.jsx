import PropTypes from "prop-types";

export default function UserDetail({ prop }) {
  return (
    <div key={prop.id}>
      <p>ID: {prop.id}</p>
      <p>Username: {prop.username}</p>
      <p>Email: {prop.email}</p>
    </div>
  );
}


UserDetail.propTypes = {
  prop: PropTypes.shape({
    id: PropTypes.number.isRequired,
    username: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};
