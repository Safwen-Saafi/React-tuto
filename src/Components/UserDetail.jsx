import PropTypes from "prop-types";

export default function UserDetail({ user }) {
  return (
    <div key={user.id}>
      <p>ID: {user.id}</p>
      <p>Username: {user.username}</p>
      <p>Email: {user.email}</p>
    </div>
  );
}


UserDetail.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    username: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};
