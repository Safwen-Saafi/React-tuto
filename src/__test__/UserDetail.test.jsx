import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import UserDetail from '../Components/Props/UserDetail';
describe('UserDetail', () => {
  it('should render user details', () => {
    // // Create mock props that match the PropTypes requirements
    const mockProp = {
      id: 1,
      username: "testuser",
      email: "test@example.com"
    };

    render(<UserDetail prop={mockProp}/>);
    // expect(result.container).toMatchSnapshot();
    expect(screen.getByText('Username: testuser')).toBeInTheDocument();
  
  });
});
