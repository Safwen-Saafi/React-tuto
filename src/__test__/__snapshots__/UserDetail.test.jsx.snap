// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserDetail > should render user details 1`] = `
<div>
  <div>
    <p>
      ID: 
      1
    </p>
    <p>
      Username: 
      testuser
    </p>
    <p>
      Email: 
      test@example.com
    </p>
  </div>
</div>
`;
